src/block/TallGrass.php

```diff
@@ -35,4 +39,30 @@ private function canBeSupportedAt(Block $block) : bool{
 		$supportBlock = $block->getSide(Facing::DOWN);
 		return $supportBlock->hasTypeTag(BlockTypeTags::DIRT) || $supportBlock->hasTypeTag(BlockTypeTags::MUD);
 	}
-}
+
+	public function onInteract(Item $item, int $face, Vector3 $clickVector, ?Player $player = null, array &$returnedItems = []) : bool{
+		if ($this->getSide(Facing::UP)->getTypeId() !== BlockTypeIds::AIR) {
+			return false;
+		}
+
+		$world = $this->position->getWorld();
+		if ($item instanceof Fertilizer) {
+			$doubleVariant = match ($this->getTypeId()) {
+				BlockTypeIds::TALL_GRASS => VanillaBlocks::DOUBLE_TALLGRASS(),
+				BlockTypeIds::FERN => VanillaBlocks::LARGE_FERN(),
+				default => VanillaBlocks::DOUBLE_TALLGRASS()
+			};
```

Comment:
It'd be better if we could add a `Closure() : DoublePlant` parameter to the constructor instead of hardcoding the accepted IDs here. That way we could just pass `fn() => VanillaBlocks::WHATEVER()` to the constructor when registering the block.

Reply:
If you're using PhpStorm you can do use the `Project Default` code style provided in the `.idea` folder (it should show up automatically). `php-cs-fixer` doesn't have a rule to fix the braces spacing currently.

Don't worry too much about it, a lot of contributors have issues with it because PM uses a non-standard code style :)